### PR TITLE
feat(api):[TRI-339] introduce depth in recursive processing

### PR DESCRIPTION
**Pull request by ds-ext-kmassalski**
I need feedback if there is a need to also serialize depth in AASTransferProcessDeserializer?

### CODE REVIEW

File: irs-api/src/main/java/net/catenax/irs/aaswrapper/job/AASTransferProcess.java [hunk 1]
```diff
@@ -57,7 +59,7 @@ public AASTransferProcess deserialize(final JsonParser jsonParser,
             final ObjectCodec codec = jsonParser.getCodec();
             final JsonNode treeNode = codec.readTree(jsonParser);
             final String idValue = treeNode.get("id").textValue();
-            return new AASTransferProcess(idValue);
+            return new AASTransferProcess(idValue, null);
```
Comment (ds-ext-kmassalski) [inline irs-api/src/main/java/net/catenax/irs/aaswrapper/job/AASTransferProcess.java:62]: Need feedback what to do here

File: irs-api/src/main/java/net/catenax/irs/controllers/IrsController.java [hunk 2]
```diff
@@ -156,10 +156,10 @@ public Job cancelJobById(
             })
     @GetMapping("/jobs")
     public List<UUID> getJobsByJobState(
-            final @Valid @ParameterObject @Parameter(description = "Requested job states.", in = QUERY,
+             @Valid @ParameterObject @Parameter(description = "Requested job states.", in = QUERY,
                     explode = Explode.FALSE, array = @ArraySchema(
                     schema = @Schema(implementation = JobState.class))) @RequestParam(
-                    value = "jobStates", required = false) List<JobState> jobStates) {
+                    value = "jobStates", required = false, defaultValue = "") final List<JobState> jobStates) {
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/controllers/IrsController.java:162]: Why the empty default?

File: irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceTest.java [hunk 3]
```diff
@@ -31,8 +43,37 @@ class IrsItemGraphQueryServiceTest {
     private IrsItemGraphQueryService service;
 
     @Test
-    void registerItemJob() {
-        assertTrue(true);
+    @Disabled // its not consistent before TRI-390 not resolved
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceTest.java:46]: Put a note into TRI-390 to enable this test again after the fix :)

File: irs-api/src/test/java/net/catenax/irs/controllers/IrsControllerTest.java [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.BDDMockito.given;
-import static org.mockito.Mockito.when;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/controllers/IrsControllerTest.java:29]: Usage of shaded object mapper - you might want to use the real dependency here

File: irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceTest.java [hunk 5]
```diff
@@ -31,8 +43,37 @@ class IrsItemGraphQueryServiceTest {
...
+
+        // then
+        given().ignoreException(EntityNotFoundException.class)
+           .await()
+           .atMost(10, TimeUnit.SECONDS)
+           .until(() -> getRelationshipsSize(registeredJob.getJobId()), equalTo(expectedRelationshipsSizeFullTree));
+    }
+
+    @Test
+    @Disabled // its not consistent before TRI-390 not resolved
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceTest.java:63]: Put a note into TRI-390 to enable this test again after the fix :)

**Review by ds-jkreutzfeld — CHANGES_REQUESTED** [review]

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**